src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -422,6 +422,14 @@ robocopy /S /NP %EXECUTION_DIR%int\%(Identity)\ %EXECUTION_DIR%native\
       <PostExecutionTestCommandLines Include="move $(UAP_Results_Path)$(XunitTestAssembly).xml .\$(XunitResultsFileName)" />
     </ItemGroup>
 
+    <!-- If the PreExecutionTestScript property is set, then it should be set to the full path to a script that will be directly incorporated
+         into the generated runtests.cmd script, immediately before the test is run. This can be used to set a number of JIT stress modes,
+         for example.
+     -->
+    <ItemGroup Condition="'$(PreExecutionTestScript)'!=''">
+      <TestCommandLines Include="$([System.IO.File]::ReadAllText('$(PreExecutionTestScript)'))" />
```

Comment:
Other things in `TestCommandLines` have been set at this point (https://github.com/BruceForstall/buildtools/blob/cc9f62ea0415b69e38a5e12093caec4ad1bfb729/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L408) so it won't happen before everything else.

Reply:
@crummel This was actually "by design" -- I didn't want anything (except perhaps environment variable settings) to come between the "PreExecutionTestScript" and the running of the test command itself. In particular, I don't want any infrastructure managed code to run with these settings. E.g., "mklink" -- is it managed code? Also, it looks like there are some other modes, like `.NET Native` that invoke, e.g., ILC, and I didn't want to be before that.

Would you prefer instead that I move this up to be the very first of the `<TestCommandLines>`?

Reply:
Ah, that makes sense.  I'm okay with it like this, would you mind adding a comment to that effect?